update photonlib + add some simulation

jack60612 opened the pull request.

Commit:
update photonlib

Commit:
fix formatting

Commit:
update photon vision code

Commit:
update to logged selection

sviridovt commented on src/main/java/frc/robot/subsystems/DriveSubsystem.java:
```diff
@@ -345,7 +345,8 @@ public void balanceCorrection(double gyroPitchAngle) {
 
   public void driveToRelativePosition(double targetPosition) {
-    if (targetPosition < 0.1) { // less then 0.1 meters, do nothing, pervents feedback loop
+    if (Math.abs(targetPosition)
```
Nit, formatting

Commit:
convert all auto movement to on the fly path creation

Commit:
multi cam yay

Iooob committed:
Added simulated motors

Commit:
add gyro sim support

Iooob committed:
Add rest simulated motors

- Create simulated motors for flywheel simulated
- Create simulated motors for lifter subsystem

Iooob committed:
Merge branch '2025updatesv2' of https://github.com/MiamiBeachBots/base-bot into 2025updatesv2

Commit:
add note

Iooob committed:
Created Elevator Subsystem + Simulated Encoders

- Elevator Subsystem (TODO: Update values)
- Added simulated encoders in Flywheel subsystem

Commit:
add simulation to drivetrain

Iooob committed:
Change to Elevator Feed

Iooob committed:
Add elevator command

Commit:
add simulation to elevator

Commit:
Update ElevatorSubsystem.java

Iooob committed:
Created arm subsystem and command

Iooob committed:
Changes

Commit:
fix amp reading

Commit:
working numbers

Commit:
Update ArmSubsystem.java

Commit:
camera sim???

Iooob committed:
Working sim???????

Iooob committed:
Created Turntable subsystem + command

- Need to setup simulator

Iooob committed:
Fix Simulated Gyro

- Still need to correct PID and other values in new files for it to simulate "properly"

Commit:
make physics engine

Commit:
Update DriveConstants.java

Commit:
no drive cam

jack60612 commented on src/main/java/frc/robot/TurntableSim.java:
```diff
@@ -0,0 +1,127 @@
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot;
```
not sure, this should be the only one

Iooob replied:
resolved

jack60612 commented on src/main/java/frc/robot/commands/AimCommand.java:
```diff
@@ -39,34 +49,65 @@ public void initialize() {}
+              targetPoses.add(
+                  new Pose2d(
+                      robotToTarget2d.getTranslation().getX() / 2,
+                      robotToTarget2d.getTranslation().getY(),
```
its just to give us a 2nd pose, as the on the fly planning refuses to use one

Iooob replied:
resolved

jack60612 commented on src/main/java/frc/robot/commands/AimCommand.java:
```diff
@@ -39,34 +49,65 @@ public void initialize() {}
+              // update the pose
+              robotToTarget2d = newTargetPose;
+              // Create list of target poses
+              // One at halfway to target, one at the target
```
we basically just need more then 1 pose essentially.

jack60612 commented on src/main/java/frc/robot/commands/ArmCommand.java:
```diff
@@ -0,0 +1,39 @@
+import frc.robot.subsystems.ArmSubsystem;
+
+/** An Arm command that uses the Arm subsystem. */
+public class ArmCommand extends Command {
```
no, the extender thing

jack60612 replied:
claw will only be in 2025 code because thats completely diff every year.

jack60612 commented on src/main/java/frc/robot/subsystems/DriveSubsystem.java:
```diff
@@ -326,83 +342,49 @@ public void tankDrive(double leftSpeed, double rightSpeed) {
+            DriveConstants.OnTheFly.kPathConstraints,
+            new IdealStartingState(0, desiredPoses.get(0).getRotation()),
+            new GoalEndState(0, desiredPoses.get(desiredPoses.size() - 1).getRotation()));
+    path.preventFlipping = true;
```
it tells it to not flip it based on the alliance, we will add comments

sviridovt replied:
Fair enough, to be fair this was more a joke than anything. Not an ask, sorry should have been more clear

jack60612 replied:
> Fair enough, to be fair this was more a joke than anything. Not an ask, sorry should have been more clear

oh I thought you were making a joke that you had no clue what was going on.

Iooob replied:
resolved, added comments

Iooob commented on src/main/java/frc/robot/commands/AimCommand.java:
```diff
@@ -39,34 +49,65 @@ public void initialize() {}
+    // TODO: update offset here
+    // robotoffset = ....
+    Optional<PhotonPipelineResult> CamResult = m_cameraSubsystem.targetingCamera1Result;
     // will not work if cam is defined incorrectly, but will not tell you
```
we should leave that comment as a reference when making changes in the future

Iooob committed:
Create simulation subpackage

- Moved TurntableSim into SImulation subpackage

Iooob committed:
Change constants to camelcase

Iooob commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -69,4 +77,53 @@ public static enum Mode {
+  public static final class PoseCamera1 {
+    public static final String name = "pose1";
+    // XYZ
+    private static final double XLocation = Units.inchesToMeters(6);
+    private static final double YLocation = Units.inchesToMeters(9.3);
+    private static final double ZLocation = Units.inchesToMeters(10.5);
```
resolved

Iooob committed:
Fixed intermediate pose at aimCommand

Iooob committed:
Formatting

- Snaked cased CAN constants
- Fixed rest of constants

Iooob commented on src/main/java/frc/robot/subsystems/ArmSubsystem.java:
```diff
@@ -0,0 +1,210 @@
+  private final double kMinAngleRads = 0.01;
+  private final double kStartingAngleRads = kMinAngleRads + 0.01;
+  private final double kArmLengthMeters = 0.1;
+  private final double kjKgMetersSquared =
+      0.1; // The moment of inertia of the arm; can be calculated from CAD software.
```
resolved

Iooob commented on src/main/java/frc/robot/subsystems/ArmSubsystem.java:
```diff
@@ -0,0 +1,210 @@
+
+  public ArmSubsystem() {
+    // Create Arm motor
+    m_Motor = new SparkMax(CANConstants.MOTORARMMAINID, SparkMax.MotorType.kBrushless);
```
resolved

Iooob committed:
extracted camera processing to own function

Iooob commented on src/main/java/frc/robot/commands/AimCommand.java:
```diff
@@ -39,34 +49,65 @@ public void initialize() {}
-      SmartDashboard.putNumber("CameraTargetPitch", 0.0);
-      m_driveSubsystem.tankDrive(0, 0);
+    CamResult.ifPresentOrElse(
+        result -> {
```
resolved

Iooob committed:
Javadocs

- Added javadocs to functions in  CameraSubsystem, Aimcommand, and DriveSubsystem
- Fixed javadocs for other subsystems

jack60612 commented on src/main/java/frc/robot/subsystems/CameraSubsystem.java:
```diff
@@ -6,83 +6,156 @@
+  private final PhotonCameraSim targetingCamera1Sim;
 
   /** Creates a new CameraSubsystem. */
   public CameraSubsystem(DriveSubsystem d_subsystem) {
```
I think it will change a lot once we get it working with real cameras.

jack60612 replied:
so i think we should do it later

Iooob committed:
Formatted navgrid & simulatiom ui

Iooob committed:
Format simui again

jack60612 reviewed (commented):
looks good in general now

Commit:
vision model correction

Iooob reviewed (dismissed):
lgtm

Commit:
fix cam code

Commit:
fix format
